analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Sonar/LocksShouldBeReleased.Monitor.cs

```
@@ -0,0 +1,208 @@
+        }
+
+        public void DoSomethingWithMonitor10(bool b)
+        {
+            bool isAcquired = false;
+            Monitor.Enter(obj, ref isAcquired); // FN
+            if (b)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor11()
+        {
+            if (Monitor.TryEnter(obj, 42))// FN
+            {
+            }
+            else
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor12(bool b)
+        {
+            bool isAcquired = false;
+            Monitor.TryEnter(obj, 42, ref isAcquired); // FN
+            if (b)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor13()
+        {
+            if (Monitor.TryEnter(obj, new TimeSpan(42)))// FN
+            {
+            }
+            else
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor14(bool b)
+        {
+            bool isAcquired = false;
+            Monitor.TryEnter(obj, new TimeSpan(42), ref isAcquired); // FN
+            if (b)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor15()
+        {
+            bool isAcquired = Monitor.TryEnter(obj, 42); // FN
+
+            if (isAcquired)
+            {
+            }
+            else
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor16()
+        {
+            bool isAcquired = Monitor.TryEnter(obj, 42); // Compliant
+
+            if (isAcquired)
+            {
+                Monitor.Exit(obj);
+            }
+        }
+
+        public void DoSomethingWithMonitor17(bool b)
+        {
+            Monitor.Enter(obj); // FN
+
+            if (b)
+            {
+                throw new Exception();
```

Let's add way more `throw` scenarios with various stuff like throw InvalidOperationException unconditionally, catch it, exit lock ONLY there (Compliant)